src/Module/Calendar/Event/API.php

```diff
@@ -0,0 +1,277 @@
+				System::httpExit($this->t('Event title and start time are required.'));
+				return;
+			} else {
+				$this->sysMessages->addNotice($this->t('Event title and start time are required.'));
+				$this->baseUrl->redirect($redirectOnError);
+			}
+		}
+
+		$self = Contact::getPublicIdByUserId($uid);
+
+		$aclFormatter = $this->aclFormatter;
+
+		if ($share) {
+			$user = User::getById($uid, ['allow_cid', 'allow_gid', 'deny_cid', 'deny_gid']);
+			if (empty($user)) {
+				$this->logger->warning('Cannot find user for an event.', ['uid' => $uid, 'event' => $eventId]);
+				$this->response->setStatus(500);
+				return;
+			}
+
+			$strAclContactAllow = isset($request['contact_allow']) ? $aclFormatter->toString($request['contact_allow']) : $user['allow_cid'] ?? '';
+			$strAclGroupAllow   = isset($request['group_allow']) ? $aclFormatter->toString($request['group_allow']) : $user['allow_gid'] ?? '';
+			$strContactDeny     = isset($request['contact_deny']) ? $aclFormatter->toString($request['contact_deny']) : $user['deny_cid'] ?? '';
+			$strGroupDeny       = isset($request['group_deny']) ? $aclFormatter->toString($request['group_deny']) : $user['deny_gid'] ?? '';
+
+			$visibility = $request['visibility'] ?? '';
+			if ($visibility === 'public') {
+				// The ACL selector introduced in version 2019.12 sends ACL input data even when the Public visibility is selected
+				$strAclContactAllow = $strAclGroupAllow = $strContactDeny = $strGroupDeny = '';
+			} else if ($visibility === 'custom') {
+				// Since we know from the visibility parameter the item should be private, we have to prevent the empty ACL
+				// case that would make it public. So we always append the author's contact id to the allowed contacts.
+				// See https://github.com/friendica/friendica/issues/9672
+				$strAclContactAllow .= $aclFormatter->toString($self);
+			}
+		} else {
+			$strAclContactAllow = $aclFormatter->toString($self);
+			$strAclGroupAllow   = $strContactDeny = $strGroupDeny = '';
+		}
+
+		$datarray = [
+			'start'     => $start,
+			'finish'    => $finish,
+			'summary'   => $summary,
+			'desc'      => $desc,
+			'location'  => $location,
+			'type'      => $type,
+			'nofinish'  => $noFinish,
+			'uid'       => $uid,
+			'cid'       => $cid,
+			'allow_cid' => $strAclContactAllow,
+			'allow_gid' => $strAclGroupAllow,
+			'deny_cid'  => $strContactDeny,
+			'deny_gid'  => $strGroupDeny,
+			'id'        => $eventId,
+		];
+
+		if (intval($request['preview'])) {
+			System::httpExit(Event::getHTML($datarray));
+			return;
+		}
+
+		$eventId = Event::store($datarray);
+
+		$newItem = Event::getItemArrayForId($eventId, [
+			'network'   => Protocol::DFRN,
+			'protocol'  => Conversation::PARCEL_DIRECT,
+			'direction' => Conversation::PUSH
+		]);
+		if (Item::insert($newItem)) {
+			$uriId = (int)$newItem['uri-id'];
+		} else {
+			$uriId = 0;
+		}
+
+		if (!$cid && $uriId) {
+			Worker::add(Worker::PRIORITY_HIGH, "Notifier", Delivery::POST, $uriId, $uid);
```

Comment:
Please convert the remaining double quotes to single.
```suggestion
			Worker::add(Worker::PRIORITY_HIGH, 'Notifier', Delivery::POST, $uriId, $uid);
```